src/java/test/org/apache/zookeeper/test/NettyLocalSuiteTest.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zookeeper.test;
+
+import org.junit.runners.Suite;
+
+/**
+ * Run tests with: Netty Client against Netty server
+ */
+@Suite.SuiteClasses({
```

Comment:
@afine I'm sorry it is not possible for old style sockets. We should provide  a full custom SocketImpl.

This change will help testing apps which use ZooKeeper and usually launch only a single node embedded ZK server inside the JVM which is running the unit test.

I think that a switch to Netty for the server-to-server communications will be a good enhancement for the future. I saw recent work about SSL on server-to-server, using Netty it would have been very simple.

I can create a JIRA for Netty on server-to-server, eventually I can work on a proposal in the near future